src/state-summary/state-card-display.html

```diff
@@ -39,8 +40,51 @@
     };
   }
 
-  computeStateDisplay(stateObj) {
-    return window.hassUtil.computeStateState(stateObj);
+  computeStateDisplay(haLocalize, stateObj) {
+    if (!stateObj._stateDisplay) {
+      const domain = window.hassUtil.computeDomain(stateObj);
+      if (domain === 'binary_sensor') {
+        // Try device class translation, then default binary sensor translation
+        stateObj._stateDisplay =
+          haLocalize(`state.${domain}.${stateObj.attributes.device_class}`, stateObj.state)
+          || haLocalize(`state.${domain}.default`, stateObj.state);
+      } else if (stateObj.attributes.unit_of_measurement) {
+        stateObj._stateDisplay = stateObj.state + ' ' + stateObj.attributes.unit_of_measurement;
+      } else if (domain === 'input_datetime') {
+        let date;
+        if (!stateObj.attributes.has_time) {
+          date = new Date(
+            stateObj.attributes.year,
+            stateObj.attributes.month - 1,
+            stateObj.attributes.day
+          );
+          stateObj._stateDisplay = window.hassUtil.formatDate(date);
+        } else if (!stateObj.attributes.has_date) {
+          date = new Date(
+            1970, 0, 1,
+            stateObj.attributes.hour,
+            stateObj.attributes.minute
+          );
+          stateObj._stateDisplay = window.hassUtil.formatTime(date);
+        } else {
+          date = new Date(
+            stateObj.attributes.year, stateObj.attributes.month - 1,
+            stateObj.attributes.day, stateObj.attributes.hour,
+            stateObj.attributes.minute
+          );
+          stateObj._stateDisplay = window.hassUtil.formatDateTime(date);
+        }
+      } else if (domain === 'zwave' && ['initializing', 'dead'].includes(stateObj.state)) {
```

Comment:
For proper i18n we should have separate messages for with and without query_stage. (One can have the other as `[%key] ` by default.

Reply:
What do you think for the schema, something like this? I personally don't think we should be including keys we're not going to use, unless I'm not understanding what you mean.

```json
"zwave": {
  "default": {
      "initializing": "Initializing",
      "dead": "Dead",
      "sleeping": "Sleeping",
      "ready": "Ready"
  },
  "query_stage": {
      "initializing": "[%key:state::zwave::initializing%] ({query_stage})",
      "dead": "[%key:state::zwave::dead%] ({query_stage})",
      "sleeping": "[%key:state::zwave::sleeping%] ({query_stage})",
      "ready": "[%key:state::zwave::ready%] ({query_stage})"
  }
}
```

Reply:
Yes, this is what I meant. Except I don't think we need query_stage for ready and sleeping.

So maybe:
```json
"zwave": {
  "default": {
      "initializing": "Initializing",
      "dead": "Dead",
      "sleeping": "Sleeping",
      "ready": "Ready"
  },
  "query_stage": {
      "initializing": "[%key:state::zwave::default::initializing%] ({query_stage})",
      "dead": "[%key:state::zwave::default::dead%] ({query_stage})",
  }
}
```

Reply:
👍 Sounds good